spanner/r2dbc/src/test/java/com/example/spanner/r2dbc/R2dbcSampleApplicationIT.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner.r2dbc;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.r2dbc.core.DatabaseClient;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import reactor.core.publisher.Mono;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+public class R2dbcSampleApplicationIT {
+
+  @DynamicPropertySource
+  static void registerProperties(DynamicPropertyRegistry registry) {
+    int rand = Math.abs((new Random()).nextInt());
+    registry.add("project", () -> ServiceOptions.getDefaultProjectId());
+    registry.add("database", () -> "r2dbc-docs-testdb-" + rand);
+
+    assertNotNull("Please provide spanner.test.instance environment variable",
+        System.getProperty("spanner.test.instance"));
+    registry.add("instance", () -> System.getProperty("spanner.test.instance"));
+  }
+
+  @Value("${database}")
+  String databaseName;
+
+  @Value("${instance}")
+  String instance;
+
+  @Autowired
+  private WebTestClient webTestClient;
+
+  @Autowired
+  DatabaseClient databaseClient;
+
+  DatabaseAdminClient dbAdminClient;
+
+  // setup/teardown cannot be static because then properties will not be injected yet
```

Comment:
Not a huge deal, but I this might be possible. Found this I believe similar issue https://stackoverflow.com/questions/17353327/populating-spring-value-during-unit-test

Reply:
Static setup/teardown is in itself possible, and populating spring values during a test is possible (this test does it), but Spring does not support static `@Value` injection, so the combination of the two is not (without ugly hackarounds).